server/plugin.go

```diff
@@ -113,13 +120,42 @@ func (p *Plugin) OnActivate() error {
 		return mutateSampleLabels(l, target, *p.configuration.HonorTimestamps, []*relabel.Config{})
 	}
 
+	ticker := time.Tick(time.Duration(scrapeInterval) * time.Second)
+	p.closeChan = make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-p.closeChan:
+				return
+			case <-ticker:
+				cfg, err := p.configuration.Clone()
```

Comment:
This can potentially be costly. Every `Clone` is a json Marshal/Unmarshal. Wondering if we can ensure to not modify the config in the `scrapeFn`? It seems like we are simply reading some config fields in there.